server.go

```diff
@@ -994,3 +997,34 @@ var _ trace.ClientBackend = &internalTraceBackend{}
 func CalculateTickDelay(interval time.Duration, t time.Time) time.Duration {
 	return t.Truncate(interval).Add(interval).Sub(t)
 }
+
```

Comment:
Maybe more at home in config_parse.go or similar? (nit)

Reply:
I do agree that server.go is getting pretty long, but at the moment, config_parse.go only contains methods on the Config struct. I didn't make this a method of the config struct because it makes it easier to test, and also there's no reason to couple the two.